examples/simple.yml

```diff
@@ -9,14 +9,14 @@
       ssh_user: root
       ssh:
         Compression: true
-        GSSAPIAuthentication: no
+        GSSAPIAuthentication: false
         ControlMaster: auto
         ControlPath: ~/.ssh/.cm%C
         Match:
           - Condition: "final all"
-            GSSAPIAuthentication: yes
+            GSSAPIAuthentication: true
         Host:
           - Condition: example
             Hostname: example.com
             User: somebody
-      ssh_ForwardX11: no
+      ssh_ForwardX11: false
```

Comment:
I would probably like to keep "yes"/"no" here, as these are the keywords of the ssh configuration file (and sometimes can have some third value) so using true/false might look unexpected. Would putting these into the quotes help to workaround the lint warning?

Reply:
> I would probably like to keep "yes"/"no" here, as these are the keywords of the ssh configuration file (and sometimes can have some third value) so using true/false might look unexpected. Would putting these into the quotes help to workaround the lint warning?

Are these values supposed to be strings or booleans?  If they are suppposed to be strings, then you must quote `"yes"` and `"no"` - otherwise, yaml/ansible will convert `yes` to boolean `True` and `no` to boolean `False`.

Reply:
> > I would probably like to keep "yes"/"no" here, as these are the keywords of the ssh configuration file (and sometimes can have some third value) so using true/false might look unexpected. Would putting these into the quotes help to workaround the lint warning?
> 
> Are these values supposed to be strings or booleans? If they are suppposed to be strings, then you must quote `"yes"` and `"no"` - otherwise, yaml/ansible will convert `yes` to boolean `True` and `no` to boolean `False`.

@Jakuje https://github.com/linux-system-roles/ssh/blob/master/templates/ssh_config.j2#L2
Doesn't this code already convert boolean `True` to the string `"yes"` and boolean `False` to the string `"no"`?  Also, all of the regression/integration tests are passing with my code.

Reply:
Yes, it is being converted to "yes", "no" strings in this role automatically. What I am referring to is mostly naming confusion in comparison with the stuff you would write into the ssh_config.
The OpenSSH should accept also the "true"/"false" strings in the end, but this is totally undocumented as every documentation/examples mention just "yes"/"no":
https://github.com/openssh/openssh-portable/blob/b0b58222c7cc62efd8212c4fb65a545f58ebb22d/readconf.c#L810
But thinking about it more, using the ansible true/false should not be a so huge issue so lets disregard my comment.